Sources/Pandora/Pandora.m

```diff
@@ -349,8 +349,14 @@ - (BOOL) createStation: (NSString*)musicId {
     Station *s = [self parseStationFromDictionary:result];
     NSMutableDictionary *dict = [NSMutableDictionary dictionary];
     dict[@"station"] = s;
-    [stations addObject:s];
-    [Station addStation:s];
+    
+    /* Add the station internally if it doesn't exist */
+    NSUInteger stationIndex = [stations indexOfObject:s];
+    if (stationIndex > [stations count]) {
+      [stations addObject:s];
+      [Station addStation:s];
+    }
+    
```

Comment:
@nriley I've cleaned this up a bit, the functionality is the same. Basically the object of this is to skip the addObject and addStation if the station already exists so that we don't get a duplicate in the station list.